### PR TITLE
Improved workflows

### PR DESCRIPTION
**Pull request by BySwiizen**
1. Added Gradle actions.
2. Improved the JDK set up.
3. Configuring Git to avoid small errors.
4. Fix names for more consistency.
5. Using a specific version of Ubuntu before the latest version arrives which can cause bugs.
6. Simple cleanup.
7. Add comments.
8. Update dependencies.

### CODE REVIEW

File: .github/workflows/gradle.yml [hunk 1]
```diff
@@ -1,27 +1,51 @@
-name: Build with Gradle
+name: Build Gradle
 
-on: [ push, pull_request ]
 
-permissions:
-  contents: read
```
Comment (datatags) [inline .github/workflows/gradle.yml:6]: Why remove this?

File: .github/workflows/gradle.yml [hunk 2]
```diff
@@ -1,27 +1,51 @@
...
+        java-package: jdk
+        architecture: x64
+        cache: gradle
+
+    - name: Configure Git
+      run: git config --global user.name "Github Actions" && git config --global user.email "actions@github.com"
+
+    - name: Make Gradlew executable
+      run: chmod +x ./gradlew
+
```
Comment (datatags) [inline .github/workflows/gradle.yml]:
```suggestion
```
I think I'd rather not have this, that way if I do ever accidentally break the permissions on `gradlew` I immediately know and can fix it for everyone who clones the repository

**Review by datatags — CHANGES_REQUESTED** [review]

File: .github/workflows/gradle.yml (hunk 1 above)
Comment (BySwiizen) [reply .github/workflows/gradle.yml:6]: It's useless.

Comment (datatags) [reply .github/workflows/gradle.yml:6]: It enforces [principle of least privilege](https://en.wikipedia.org/wiki/Principle_of_least_privilege) though, so I'd like to keep it

**Review by datatags — APPROVED** [review]
Thanks!

**Pull request merged**